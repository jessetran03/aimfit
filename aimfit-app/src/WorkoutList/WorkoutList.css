* {
  margin: 0;
  padding: 0;
}

body {
  font-family: sans-serif;
  font-size: 20px;
  text-align: center;
}

header {
  padding: 20px;
  background-color: #AAAAAA;
}

h3 {
  background-color: #FFFFFF;
}

.exercises {
  background-color: #AAAAAA;
}

.exercises li {
  background-color: #FFFFFF;
  padding: 15px 0;
  margin: 10px;
  border-style: solid;
}