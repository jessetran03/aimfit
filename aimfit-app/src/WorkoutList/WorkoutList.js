import React, { Component } from 'react'
import './WorkoutList.css'
import Workout from '../Workout/Workout'
import STORE from '../STORE'

export default class WorkoutList extends Component {
  state = {
    store: STORE,
  }

  handleAddWorkout = () => {

    const workoutList = this.state.store.workouts
    const newWorkout = {
      id: '11',
      title: 'New Workout',
      exerciseIds: ['1'],
    }

    workoutList.push(newWorkout)

    this.setState({
      store: {
        workouts: workoutList
      }
    })

  };

  render() {
    const { store } = this.state;
    return (
      <>
        <section className="exercises">
          <h2>Workout List</h2>
          <ul>
            {store.workouts.map(workout => (
              <li
                key={workout.id}
              >
                <Workout
                  id={workout.id}
                  title={workout.title}
                />
              </li>
            ))}
          </ul>
          <button
            type="button"
            className="List-add-button"
            onClick={() => this.handleAddWorkout()}
          >
            + Add Workout
          </button>
        </section>
      </>
    )
  }
}

/*const { folderId } = this.props.match.params
const { notes = [] } = this.context
const notesForFolder = getNotesForFolder(notes, folderId)

export const getNotesForFolder = (notes = [], folderId) => (
  (!folderId)
    ? notes
    : notes.filter(note => note.folder_id === parseInt(folderId))
)
*/