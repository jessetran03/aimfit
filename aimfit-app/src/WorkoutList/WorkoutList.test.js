import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import WorkoutList from './WorkoutList'
import renderer from 'react-test-renderer'

it('renders WorkoutList without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <BrowserRouter>
      <WorkoutList />
    </BrowserRouter>,
    div
  )
  ReactDOM.unmountComponentAtNode(div)
})

it('render WorkoutList without data', () => {
  const wrapper = renderer.create(
    <BrowserRouter>
      <WorkoutList />
    </BrowserRouter>
  )
    .toJSON();
    expect(wrapper).toMatchSnapshot()
  })