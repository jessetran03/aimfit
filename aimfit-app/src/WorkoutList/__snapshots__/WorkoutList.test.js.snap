// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render WorkoutList without data 1`] = `
<section
  className="workouts"
>
  <h2>
    Workout List
  </h2>
  <ul />
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="workout-name-input"
    >
      Name of workout:  
    </label>
    <input
      id="workout-name-input"
      name="workout-name"
      required={true}
      type="text"
    />
    <br />
    <label
      htmlFor="workout-day-input"
    >
      Day of week:  
    </label>
    <select
      id="workout-day-select"
      name="workout-day"
    >
      <option
        value="Sunday"
      >
        Sunday
      </option>
      <option
        value="Monday"
      >
        Monday
      </option>
      <option
        value="Tuesday"
      >
        Tuesday
      </option>
      <option
        value="Wednesday"
      >
        Wednesday
      </option>
      <option
        value="Thursday"
      >
        Thursday
      </option>
      <option
        value="Friday"
      >
        Friday
      </option>
      <option
        value="Saturday"
      >
        Saturday
      </option>
    </select>
    <br />
    <button
      className="List-add-button"
      type="submit"
    >
      + Add Workout
    </button>
  </form>
</section>
`;
